OSDOCS-9171#Add SMB CIFS CSI Driver Operator (#75920)
openshift/openshift-docs: modules/persistent-storage-csi-smb-cifs-create-sc.adoc

// Module included in the following assemblies:
//
// * storage/container_storage_interface/persistent-storage-csi-smb-cifs.adoc

:_mod-docs-content-type: PROCEDURE
[id="persistent-storage-csi-smb-cifs-create-sc_{context}"]
= Creating a storage class for SMB CIFS

After installing the Operator, you should create a storage class for dynamic provisioning of Server Message Block (SMB) protocol Common Internet File System (CIFS) dialect volumes.

.Prerequisites

> * You are logged in to the running {product-title} cluster.

Reviewer: Let's two more prerequisites:

> * You have SMB server installed and know its:
>  - hostname
>  - share name
>  - username/password
> * You created a Secret called `smbcreds` in the namespace `samba-server` and put username and password as `stringData` there.

Author: I added the 1st prerequiste. I have the 2nd one as part of the procedure.

.Procedure
To create a storage class:

* Create a storage class using the following example YAML file:
+
[source,yaml]
.Example YAML file
--
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: samba
provisioner: smb.csi.k8s.io
parameters:

>   source: //samba-server.samba-server.svc.cluster.local/share

Reviewer: I'd suggest some explanation for this line:

> This assumes that Samba server is installed somewhere and reachable from the cluster by the hostname `samba-server.samba-server.svc.cluster.local`. And the server is configured to have the path `/share` among exported shares.

Reviewer 2: yes +1 let's be exhaustive on the pre-requisites

  csi.storage.k8s.io/provisioner-secret-name: smbcreds
  csi.storage.k8s.io/provisioner-secret-namespace: samba-server
  csi.storage.k8s.io/node-stage-secret-name: smbcreds

>   csi.storage.k8s.io/node-stage-secret-namespace: samba-server

Reviewer: These four lines assume that the user put username/password for access to Samba sever to a Secret with name `smbcreds` in the namespace `samba-server`:

```
apiVersion: v1
kind: Secret
metadata:
  name: smbcreds
  namespace: samba-server
stringData:
  username: <put_username_here>
  password: <put_password_here>
```

I think we need such an explanation about `smbcreds` and `samba-server` to make the doc useful.

Reviewer 2: +1 we need a more generic snippet and detailed explanations below

reclaimPolicy: Delete
volumeBindingMode: Immediate
mountOptions:

>   - mfsymlinks

Reviewer 3: The suggested moutOptions should be as follows
```suggestion
  - dir_mode=0777
  - file_mode=0777
  - noperm
  - mfsymlinks
  - cache=strict
  - noserverino
```
The suggested moutOptions should be

--

> 

Reviewer 4: You may want to add a second step here showing how to apply the yaml file, such as step 3 in [creating a compute machine set](https://docs.openshift.com/container-platform/4.15/machine_management/creating_machinesets/creating-machineset-azure.html#machineset-creating_creating-machineset-azure).

// Module included in the following assemblies:
//
// * storage/container_storage_interface/persistent-storage-csi-smb-cifs.adoc

:_mod-docs-content-type: PROCEDURE
[id="persistent-storage-csi-smb-cifs-create-sc_{context}"]
= Creating a storage class for SMB CIFS

> After installing the Operator, you should create a storage class for dynamic provisioning of Server Message Block (SMB) protocol Common Internet File System (CIFS) dialect volumes.

Reviewer 2: I would like to see a second section that explains how to manually create a PV that points to an existing share.
This will be a common use case where OCP consume and existing CIFS share that already contains data.
https://github.com/kubernetes-csi/csi-driver-smb/blob/master/deploy/example/e2e_usage.md#option2-pvpvc-usage

// Module included in the following assemblies:
//
// * storage/container_storage_interface/persistent-storage-csi-smb-cifs.adoc

:_mod-docs-content-type: PROCEDURE
[id="persistent-storage-csi-smb-cifs-create-sc_{context}"]
= Creating a storage class for CIFS/SMB

After installing the Operator, you should create a storage class for dynamic provisioning of Common Internet File System (CIFS) dialect/Server Message Block (SMB) protocol volumes.

.Prerequisites
* You are logged in to the running {product-title} cluster.

> * You have SMB server installed and know its:

Reviewer 5: Per the [ISG](https://www.ibm.com/docs/en/ibm-style?topic=format-lists#lead-in-wording) we should use complete sentences to introduce lists: 
```suggestion
* You have SMB server installed and know the following information about the server:
```

Feel free to change up some of the wording if you think of a better way to phrase it.

> ** hostname
> ** share name
> ** username/password

Reviewer 5: Unless these are specific terms that need to be lowercase, I think these should be capitalized:
```suggestion
** Hostname
** Share name
** Username/password
```

> ** username/password

Reviewer 5: Suggestion to use "and" instead:
```suggestion
** username and password
```

.Procedure

> To create a storage class:

Reviewer 5: I don't think we typically introduce procedures like this after the .Procedure title, plus I think it's already clear that this is what a user is going to be doing imo. I would suggest removing this line:
```suggestion
```
